woocommerce/woocommerce
# Add `coming soon` page and setting

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds `coming soon` page when user visits the core profiler for the first time.
It also adds a page selector for `coming soon` page under WooCommerce -> Settings -> Advanced.

![Screen Shot 2024-03-20 at 1 15 57 PM](https://github.com/woocommerce/woocommerce/assets/4723145/527aa442-905c-449b-8631-3d1e0eceaecf)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. When you're redirected to the core profiler, do not proceed.
3. Open a new window and enable `launch-your-store` feature flag.
4. Refresh the core profiler page and skip it.
5. Go to `Pages` and confirm `Coming soon` page has been created.
6. Go to `WooCommerce -> Settings -> Advanced` and confirm `Coming soon` page selector shows up. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add coming soon page and its page selector

#### Comment 

</details>
